webview_panels/src/modules/documentationEditor/components/documentationPropagation/DocumentationPropagation.tsx

```
@@ -38,58 +46,155 @@ interface DownstreamColumns {
+  columnDescription: string;
+  selectedColumns: Record<string, boolean>;
+  setSelectedColumns: React.Dispatch<
+    React.SetStateAction<Record<string, boolean>>
+  >;
+  downstreamColumns: DocsItem[];
+}) => {
+  const setAllColumnsValue = (value: boolean) => {
+    setSelectedColumns(
+      downstreamColumns.reduce(
+        (acc, curr) => ({
+          ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)